helix-core/src/line_ending.rs

```diff
@@ -13,6 +13,29 @@ pub enum LineEnding {
     PS,   // U+2029 -- ParagraphSeparator
 }
 
+impl LineEnding {
+    pub fn len(&self) -> usize {
+        match self {
+            Self::Crlf => 2,
+            _ => 1,
+        }
+    }
```

Comment:
This looks like it's giving the line-ending size in chars.  Probably re-name to `len_chars()` to be clear.

(Or if you meant it to be bytes, then this isn't correct.  For example, [`LS` is 3 bytes in utf8](https://unicode-table.com/en/2028/), among others.)